File: app-java/src/main/java/tech/thdev/java_udemy_sample/data/FlickrPhoto.java
package tech.thdev.java_udemy_sample.data;

/**
 * Created by tae-hwan on 11/17/16.
 */

public class FlickrPhoto {
    public String id;
    public String secret;
    public String server;
    public int farm;
    public String dateuploaded;
    public FlickrOwner owner;
    public FlickrContent title;
    public FlickrContent description;
    public FlickrDates dates;
    public FlickrContent comments;
    public String views;
    public FlickrUrls urls;
    public String media;

    /**
     * API : <a href="https://www.flickr.com/services/api/misc.urls.html">Photo source url</a>
     * https://farm{farm-id}.staticflickr.com/{server-id}/{id}_{secret}.jpg
     */
    public String getUrl() {
        return String.format("https://farm%s.staticflickr.com/%s/%s_%s.jpg", farm, server, id, secret);
    }
}

File: old-app-kotlin/src/main/java/tech/thdev/kotlin_udemy_sample/data/FlickrPhoto.kt
package tech.thdev.kotlin_udemy_sample.data

/**
 * Created by tae-hwan on 11/13/16.
 */

data class FlickrPhoto(val id: String,
                       val secret: String,
                       val server: String,
                       val farm: Int,
                       val dateuploaded: String,
                       val owner: FlickrOwner,
                       val title: FlickrContent,
                       val description: FlickrContent,
                       val dates: FlickrDates,
                       val comments: FlickrContent,
                       val views: String,
                       val urls: FlickrUrls,
                       val media: String) {

    /**
     * API : <a href="https://www.flickr.com/services/api/misc.urls.html">Photo source url</a>
     * https://farm{farm-id}.staticflickr.com/{server-id}/{id}_{secret}.jpg
     */
    fun getImageUrl()
            = "https://farm$farm.staticflickr.com/$server/${id}_$secret.jpg"
}